CDP4IME/CDP4PluginLoader.cs

```diff
@@ -25,26 +26,35 @@ public class CDP4PluginLoader
         private const string PluginDirectoryName = "plugins";
 
         /// <summary>
+        /// The name of the bew plugins to be added to the app settings
+        /// </summary>
+        public readonly List<string> NewPlugins;
+
+        /// <summary>   
         /// Initializes a new instance of the <see cref="CDP4PluginLoader"/> class.
         /// </summary>
         public CDP4PluginLoader(IAppSettingsService<ImeAppSettings> appSettingsService)
         {
             this.DirectoryCatalogues = new List<DirectoryCatalog>();
 
             var currentPath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
-            
+
             var path = Path.Combine(currentPath, PluginDirectoryName);
             var directoryInfo = new DirectoryInfo(path);
+            this.NewPlugins = new List<string>();
 
             if (directoryInfo.Exists)
             {
                 foreach (var dir in directoryInfo.EnumerateDirectories())
                 {
-                    var fileName = Path.GetFileName(dir.FullName);
+                    var pluginName = dir.FullName.ToUpper();
+                    var locationInfo = new DirectoryInfo(pluginName);
+                    var pluginSettingsList = appSettingsService.AppSettings.Plugins.Where(x => (x.Key != null) && (x.Key.ToUpper() == locationInfo.Name)).ToList();
```

Comment:
Same changes as requested in the Addin pluginloader.

Reply:
Code has been refactored